src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigValidator.kt

```diff
@@ -14,28 +16,51 @@ class AuroraConfigValidator(val applicationId: ApplicationId,
                             val auroraConfigFields: AuroraConfigFields) {
     val logger: Logger = LoggerFactory.getLogger(AuroraConfigValidator::class.java)
 
-    fun validate() {
-        val errors = fieldHandlers.mapNotNull { e ->
+    @JvmOverloads
+    fun validate(checkUnmappedPointers: Boolean = true) {
+        val errors: List<ValidationError> = fieldHandlers.mapNotNull { e ->
             val auroraConfigField = auroraConfigFields.fields[e.name]
+            val result = e.validator(auroraConfigField?.value)
 
-            e.validator(auroraConfigField?.value)?.let {
-                ValidationError(it.localizedMessage, auroraConfigField)
+            when {
+                result == null -> null
+                auroraConfigField != null -> illegalConfigField(result.localizedMessage, auroraConfigField)
+                else -> missingConfigField(e.path, result.localizedMessage)
             }
         }
 
-        val unmappedErrors = getUnmappedPointers().flatMap { pointerError ->
-            pointerError.value.map { ValidationError("$it is not a valid config field pointer", fileName = pointerError.key) }
+        val unmappedErrors = if (checkUnmappedPointers) {
+            getUnmappedPointers().flatMap { pointerError ->
+                pointerError.value.map { invalidConfigField(it, pointerError.key) }
+            }
+        } else {
+            emptyList()
         }
 
         (errors + unmappedErrors).takeIf { it.isNotEmpty() }?.let {
             logger.debug("{}", it)
             val aid = applicationId
             throw ApplicationConfigException(
                     "Config for application ${aid.application} in environment ${aid.environment} contains errors",
-                    errors = it.mapNotNull { it })
+                    errors = it.mapNotNull { it }
+            )
         }
     }
 
+    private fun illegalConfigField(message: String, auroraConfigField: AuroraConfigField): ValidationError {
```

Comment:
The ValidationError class should probably be part of the model (not in the exceptions-file), and all three utility methods should be moved to that class and renamed for brevity; i.e. ValidationError.illegal(message, auroraConfigField). Also, the order of the message-parameter is inconsistent accross the utility methods. Should probably be the last parameter in both illegalConfigField and missingConfigField.